File: LibraryOrderCore/Models/OrderModel.cs
using LibraryOrderCore.Data.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace LibraryOrderCore.Models
{
    public class OrderModel
    {
        public int OrderId { get; set; }
        public string OrderNumber { get; set; }
        public DateTime OrderDate { get; set; }
        public bool IsContacted { get; set; }

        public string CustomerFirstName { get; set; }
        public string CustomerLastName { get; set; }
        public string CustomerEmail { get; set; }
        public string CustomerPhoneNumber { get; set; }

        public ICollection<OrderItemModel> OrderItems { get; set; }
        
    }
}

File: LibraryOrderClientApp/src/app/shared/order.model.ts
export interface IOrder {
    orderId?: number
    orderNumber: string
    orderDate: string
    isContacted: boolean
    customerFirstName: string
    customerLastName: string
    customerEmail: string
    customerPhoneNumber: string
    orderItems?: IOrderItems[]
}

export interface IOrderItems {
    orderItemId: number
    book?: IBook;
    quantity: number
    isOrdered: boolean
}

export interface IBook {
    bookId?: number
    title: string
    author: string
    editor: string
    isbn: string
}